ACCS-534: My Account: Delete a stored card in My Account (#820)
commerce-docs/microsite-commerce-storefront: src/content/docs/dropins/user-account/containers/payment-methods.mdx

import OptionsTable from '@components/OptionsTable.astro';
import Aside from '@components/Aside.astro';

> The `PaymentMethods` container lists a customer's stored payment methods (for example, vaulted cards), lets them remove a method after confirming in a dialog, and supports a compact layout for account dashboards. It uses the same card presentation as checkout-style flows via the internal `PaymentCard` component.

Reviewer: ```suggestion
The `PaymentMethods` container shows saved payment methods for the signed-in customer (for example, cards saved at checkout). **Vaulting** means Adobe Commerce stores a **token** for the card (not the full card number) so the shopper can reuse that method later. You can show a full section or a **compact** row layout next to other account widgets. The list uses the same card styling as checkout flows through the `PaymentCard` component.
```

  ![Minified layout with multiple saved cards](@images/dropins/user-account/payment-methods-minified.png)
</Diagram>

## Remove confirmation

> Removing a stored method is a **two-step** flow:

Reviewer: ```suggestion
This section describes **what the shopper sees** when they remove a **stored payment method** from the list—for example, a card they saved (vaulted) at checkout—and **what runs on the server** afterward, so you can test the flow or explain it to authors.
```

> 1. The shopper chooses **Remove** on a row. A **confirmation modal** opens (it can show a compact **PaymentCard** preview with brand, last digits, and expired state when that data is available).

Reviewer: ```suggestion
1. The shopper selects **Remove** on a row. 
```

> 2. **Cancel** closes the modal and does **not** call the API. **Remove** runs the **`deletePaymentToken`** mutation (or the equivalent path for your data source—see [Data loading](#data-loading)).

Reviewer: ```suggestion
2. A **remove confirmation** dialog opens. The dialog can show a small **PaymentCard** preview with the card brand, last digits, and expired state when that data is available.
```

> 

Reviewer: ```suggestion

The shopper then chooses one option:

- **Cancel** closes the dialog. Nothing is deleted on the server.
- **Remove** applies the deletion. When the list was loaded with GraphQL, the UI calls **`deletePaymentToken`**. When the list came only from the event bus, only the in-memory row is removed (see [Data loading](#data-loading)).

When **Remove** succeeds, the row disappears from the list and the dialog closes.
```

### Full layout

<Diagram caption="Remove payment confirmation modal (full Payments page)">
  ![Remove payment confirmation modal on a full Payments page](@images/dropins/user-account/remove-payment-confirmation-modal.png)
</Diagram>

### Minified layout

<Diagram caption="Remove payment confirmation modal (minified dashboard)">
  ![Remove payment confirmation modal in minified layout](@images/dropins/user-account/remove-payment-confirmation-modal-minified.png)
</Diagram>

> While a removal request is in progress, the modal shows a **loading** state on the confirm action, and **remove controls on all rows are disabled** until the operation finishes. Customize modal copy with the dictionary keys under **`Account.Payments.removePaymentModal`** (see [Stored payment methods (Payments)](/dropins/user-account/dictionary/#stored-payment-methods-payments)).

Reviewer: ```suggestion
While a removal request is running, the dialog shows **loading** on the confirm control. **Remove** is disabled on every row until the request finishes. 

You can change the dialog labels and body text with dictionary keys under **`Account.Payments.removePaymentModal`** (see [Stored payment methods (Payments)](/dropins/user-account/dictionary/#stored-payment-methods-payments)).
```
